src/DDSPreview.py

```diff
@@ -172,14 +176,44 @@ def setBackgroundColour(self, colour: QColor):
     def getBackgroundColour(self) -> QColor:
         return self.backgroundColour
 
-    def setChannels(self, channels: ColourChannels):
-        if isinstance(channels, ColourChannels):
-            self.channels = channels
+    def getChannelMatrix(self) -> QMatrix4x4:
+        return self.channelMatrix
+
+    def setChannelMatrix(self, matrix):
+
+        def flattenList(lst: Iterable):
+            tmp = []
+            for e in lst:
+                if isinstance(e, Iterable):
+                    tmp += flattenList(e)
+                elif isinstance(e, (int, float)):
+                    tmp += [float(e)]
+                else:
+                    raise ValueError("Can only set a matrix with numbers.")
+            return tmp
+
+        if isinstance(matrix, Iterable):
+            flattened = flattenList(matrix)
+            if len(flattened) != 16:
+                raise ValueError("Must provide exactly 16 values.")
+            matrix = QMatrix4x4(flattened)
+
+        if isinstance(matrix, QMatrix4x4):
+            self.channelMatrix = matrix
         else:
-            raise TypeError(str(channels) + " is not a valid ColourChannels object.")
+            raise ValueError("Can only set a matrix with numbers.")
```

Comment:
Can we not just do `self.channelMatrix = QMatrix4x4(matrix)` and let PyQt6/SIP handle any conversions? I don't think this lets everything that can be converted to a matrix through, and the extra supported stuff and error reporting probably aren't that useful. It's more obvious if it just accepts what people would expect it to accept, which is matrices.

Reply:
Yeah, that''s much better. Don't know why I made it so complicated.